contract/models/account_analytic_account.py

```diff
@@ -198,9 +198,19 @@ def _prepare_invoice_line(self, line, invoice_id):
         contract = line.analytic_account_id
         if 'old_date' in self.env.context and 'next_date' in self.env.context:
```

Comment:
I think It's simpler if you do this change instead:

```suggestion
        if 'old_date' in self.env.context and 'next_date' in self.env.context:
            lang_obj = self.env['res.lang']
            code = contract.partner_id.lang or self.company_id.partner_id.lang
            lang = lang_obj.search(
                [('code', '=', code)])
            date_format = lang.date_format or '%m/%d/%Y'
```

What do you think?

Reply:
Hello @cubells  
I am ok with your suggested changes.